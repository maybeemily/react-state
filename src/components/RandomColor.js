import React, { PureComponent } from 'react';

export default class RandomColor extends PureComponent {
  state = {
    color: '',
    trigger: new Date(),
  };

  randomColor = () => {
    const colorArray = ['olive', 'olivedrab', 'orange', 'orangered', 'orchid', 'navy'];
    const n = Math.floor(Math.random() * colorArray.length);

    return colorArray[n];
  }

  componentDidMount() {
    this.intervalId = setInterval(() => {
      this.setState({ color: this.randomColor(), trigger: new Date() });
    }, 500);
  }

  componentWillUnmount() {
    this.intervalId && clearInterval(this.intervalId);
  }

  render() {
    const { color } = this.state;
    const style = {
      background: color,
      width: '500px',
      height: '500px'
    };
    return <div style={style}></div>;
  }
}
