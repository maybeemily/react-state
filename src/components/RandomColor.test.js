import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('Random color component', () => {
  it('renders the random colors', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
