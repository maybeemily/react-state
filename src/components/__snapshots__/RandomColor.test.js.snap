// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random color component renders the random colors 1`] = `
<div
  style={
    Object {
      "background": "",
      "height": "500px",
      "width": "500px",
    }
  }
/>
`;
